### DISCUSSION TITLE
Configure prettier for Adonis 5

### DISCUSSION THREAD

**Discussion by Pacheco95** [adonisjs/core#1590] [Help] [answered]
I created a project with `npm init adonis-ts-app` and selected the option `Setup eslint? (y/N) · true`. This setup gives me the following `.eslintrc.json`

```json
{
  "extends": [
    "plugin:adonis/typescriptApp"
  ]
}
```

I also have configured my `.vscode/settings.json` to this
```jsonc
{
  "editor.formatOnSave": true,
  "editor.codeActionsOnSave": {
    // For ESLint
    "source.fixAll.eslint": true,
    // For TSLint
    "source.fixAll.tslint": true,
    // For Stylelint
    "source.fixAll.stylelint": true
  }
}
```
But when I save the `.ts` files, prettier seems to ignore eslint settings.

What should I do to make eslint correctly format my `.ts` files on save?

**Comment by prashanth1k** [comment]
Does Prettier ignore `eslint` but still format the file? There are many flags to play around and run amok, unfortunately. Did you try [this post](https://medium.com/@pgivens/write-cleaner-code-using-prettier-and-eslint-in-vscode-d04f63805dcd)?

**Reply by Pacheco95** [reply]
Yes, it does. And yes, I tried `"prettier.eslintIntegration": true` but VsCode shows the message `Unknown Configuration Setting` on hovering this line. Seems this command is no longer supported.
I discovered these settings in this video first comment https://www.youtube.com/watch?v=TI4v4Y8yRjw (it is in portuguese)

**Comment by fnoquiq** [comment, marked as answer]
[Try it](https://www.notion.so/Padr-es-de-projeto-com-ESLint-Prettier-e-EditorConfig-0b57b47a24724c859c0cf226aa0cc3a7) (Brazilian portuguese): 

.eslintrc.json
```ts 
{
  "extends": [
    "plugin:adonis/typescriptApp",
    "prettier/@typescript-eslint",
    "plugin:prettier/recommended"
  ],
  "plugins": [
    "prettier"
  ],
  "rules": {
    "prettier/prettier": "error",
    "no-console": "error"
  }
}
```

prettier.config.js 
```ts
module.exports = {
  trailingComma: "es5",
  semi: false,
  singleQuote: true,
  quoteProps: "consistent",
  bracketSpacing: true,
  arrowParens: "always",
  printWidth: 100,
  tabWidth: 2,
}
```

**Reply by Pacheco95** [reply]
Worked like a charm!!! Thank you very much :rocket:
